node/actors/consensus/src/replica/leader_commit.rs

```diff
@@ -1,9 +1,33 @@
 use super::StateMachine;
-use crate::{inner::ConsensusInner, replica::error::Error};
+use crate::inner::ConsensusInner;
+use anyhow::Context as _;
 use concurrency::ctx;
 use roles::validator;
 use tracing::instrument;
 
+#[derive(thiserror::Error, Debug)]
+#[allow(clippy::missing_docs_in_private_items)]
+pub(crate) enum Error {
+    #[error("bad protocol version")]
+    BadProtocolVersion,
```

Comment:
It could make sense to include the expected and observed versions here for debug purposes.

Reply:
done